desktop/core/src/desktop/loglistener.py

```diff
@@ -0,0 +1,214 @@
+      self.socket.bind(self.server_address)
+      st = os.stat(self.server_address)
+      os.chmod(self.server_address, st.st_mode | stat.S_IWOTH)
+
+    def server_activate(self):
+      self.socket.listen(self.request_queue_size)
+
+    def serve_until_stopped(self):
+      abort = 0
+      while not abort:
+        rd, wr, ex = select.select([self.socket.fileno()],
+                       [], [],
+                       self.timeout)
+        if rd:
+          self.handle_request()
+        logging._acquireLock()
+        abort = self.abort
+        logging._releaseLock()
+      self.server_close()
+
+  class Server(threading.Thread):
+    def __init__(self, rcvr, hdlr, server_address, verify):
+      super(Server, self).__init__()
+      self.rcvr = rcvr
+      self.hdlr = hdlr
+      self.server_address = server_address
+      self.verify = verify
+      self.ready = threading.Event()
+
+    def run(self):
+      server = self.rcvr(server_address='/tmp/hue.uds', handler=self.hdlr,
+                         ready=self.ready, verify=self.verify)
+      self.ready.set()
+      global _udslistener
+      logging._acquireLock()
+      _udslistener = server
+      logging._releaseLock()
+      server.serve_until_stopped()
+
+  return Server(ConfigSocketReceiver, ConfigStreamHandler, server_address, verify)
+
+def udsstopListening():
+  """
+  Stop the listening server which was created with a call to listen().
+  """
+  global _udslistener
+  logging._acquireLock()
+  try:
+    if _udslistener:
+      _udslistener.abort = 1
+      _udslistener = None
+  finally:
+    logging._releaseLock()
+
+def argprocessing(args=[], options={}):
+  parser = argparse.ArgumentParser(prog='runloglistener', description='What this program does', epilog='Text at the bottom of help')
+  parser.add_argument('-s', '--socket', dest='socket', action='store', default='/tmp/hue.uds')
+
+  opts = parser.parse_args()
+  if opts.socket:
+    options['socket'] = opts.socket
+
+def enable_logging(args, options):
+  CONF_RE = re.compile('%LOG_DIR%')
+  CONF_FILE = os.path.abspath(os.path.join(os.path.dirname(os.path.dirname(__file__)),
+                                           '..', '..', 'conf', 'gunicorn_log.conf'))
+  if os.path.exists(CONF_FILE):
+    log_dir = os.getenv("DESKTOP_LOG_DIR", "/var/log/hue")
+    raw = open(CONF_FILE).read()
+    def _repl(match):
+      if match.group(0) == '%LOG_DIR%':
+        return log_dir
+    sio = string_io(CONF_RE.sub(_repl, raw))
+    logging.config.fileConfig(sio)
+  root_logger = logging.getLogger()
+  root_logger.info("Welcome to Hue from Listener server ")
+
+class LogException(Exception):
+  def __init__(self, e):
+    super(LogException, self).__init__(e)
+    self.message = e.status.message
+
+  def __str__(self):
+    return self.message
+
+rt = None
+def signal_handler(sig, frame):
+  global rt
+  udsstopListening()
+  rt.join()
+  print("Received %s" % sig)
```

Comment:
Please also place a print statement at the start of the function. We don't want to remain stuck if there is some lock-issue/deadlock in `udsstopListening()`.

Reply:
ok